x/oracle/keeper/oracle.go

```diff
@@ -88,33 +91,28 @@ func (k Keeper) ApproveOracleRegistration(ctx sdk.Context, msg *types.MsgApprove
 
 }
 
-// validateApproveOracleRegistration checks signature
-func (k Keeper) validateApproveOracleRegistration(ctx sdk.Context, msg *types.MsgApproveOracleRegistration) error {
-
+// validateApprovalSharingOracleKey validate unique ID of ApprovalSharingOracleKey and its signature
+func (k Keeper) validateApprovalSharingOracleKey(ctx sdk.Context, approval *types.ApprovalSharingOracleKey, signature []byte) error {
 	params := k.GetParams(ctx)
-	targetOracleAddress := msg.ApprovalSharingOracleKey.TargetOracleAddress
 
 	// check unique id
-	if msg.ApprovalSharingOracleKey.UniqueId != params.UniqueId {
+	if approval.UniqueId != params.UniqueId {
 		return types.ErrInvalidUniqueID
 	}
 
-	// verify signature
-	if err := k.VerifyOracleSignature(ctx, msg.ApprovalSharingOracleKey, msg.Signature); err != nil {
-		return err
-	}
-
-	if msg.ApprovalSharingOracleKey.EncryptedOraclePrivKey == nil {
-		return fmt.Errorf("encrypted oracle private key is nil")
+	// check if the approver oracle exists
+	existing, err := k.HasOracle(ctx, approval.ApproverOracleAddress)
+	if err != nil || !existing {
+		return fmt.Errorf("failed to check if the approver oracle exists or not. address(%s)", approval.ApproverOracleAddress)
```

Comment:
How about splitting the error? It seems that the address may not be valid and the oracle may not be in the store.

Reply:
Got it. [1c69a6d](https://github.com/medibloc/panacea-core/pull/582/commits/1c69a6de0ebc97888b33d92718875dc523c84cac)

As you can see the commit, I changed the `HasOracle` function.
Because we should check if the oracle exists and is active as well. Thank you for pointing out.